### DISCUSSION TITLE
Support for servers list URL

### DISCUSSION THREAD

**Discussion by alexeys85** [AdguardTeam/AdGuardHome#3660] [Ideas]
It would be great to have support of parsing extensive list of public DNS resolvers which available in dnscrypt-resolvers github with filters like non-logging, non-censoring and so on. So instead of writing multiple lines of servers in Upstream DNS servers field it would be enough to point just one url with server list.

**Comment by ainar-g** [comment]
You can kind of already do that with the `upstream_dns_file` field in the configuration file.  See [this][1] section on the Wiki.

[1]: https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#upstreams-from-file

**Reply by alexeys85** [reply]
Yes, I do. But I can't set parameters like use only non-logging and non-filtering servers which is encoded in sdns stamp.
Also I didn't find format of that `upstream_dns_file` file.

**Reply by ainar-g** [reply]
If I recall correctly, the format is the same as the upstream input on the UI.  But you'll still need another piece of software to filter out the unnecessary ones as well as to update it every once in a while.  It's more of a workaround as opposed to a solution.

**Reply by alexeys85** [reply]
Yep, that's a workaround. Ideally it would by great to have it implemented in WEB UI!

**Comment by mgcWil** [comment]
If it were also possible to exclude certain servers, by name in the manner of DNScrypt.

To choose only certain countries, close to you, to filter also the dnssec, ipv6, protocol...

The addresses of the servers in this list can change, or even stop responding, temporarily or permanently. Adguard Home should be able to take this into account and react accordingly. 

If it could manage this for us, according to our choices, automatically, it would be great!
